docs/developer/development/ci.rst

```diff
@@ -0,0 +1,146 @@
+Continuous Testing
+==================
+
+Add a new test in the continuous integration system
+---------------------------------------------------
+
+A test as continuous integration system level is an End-to-End test that need a
+dedicated environment (with one or several machines (virtual or container)).
+
+A test that only checks a specific feature of a classic MetalK8s deployment
+should be part of PyTest BDD and not integrated as a dedicated stage in
+continuous integration system.
+
+How to choose between Pre-merge and Post-merge
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The choice really depends on the goals of this test.
+
+As a high-level view:
+
+Pre-merge:
+
+- Test is usually not long and could last less than 30 minutes.
+- Test essential features of the product (installation, expansion, building,
+  ...).
+
+Post-merge:
+
+- Test last longer (more than 30 minutes).
+- Test "non-essential" feature of the product (upgrade, downgrade, solutions,
```

Comment:
I am a little bit sceptical about `non-essential` in this sentence. Maybe we leave it at:
`Test the following features of the product (upgrade, downgrade, solutions ...).`

Reply:
Yes but the list is not exhaustif so .... 
Maybe I can just add `(not mandatory to have a working cluster)`